cmd/add.go

```diff
@@ -294,6 +315,82 @@ func fileDirIsInDirs(file string, dirSet map[string]bool) bool {
 	return false
 }
 
+// handleMeta takes the user provided meta and the backup meta inputs and
+// returns a map containing all valid metadata.
+func handleMeta(meta, reason, review, removal string) map[string]string {
+	userMeta := parseMetaString(meta)
+
+	backupMeta, err := createBackupMetadata(reason, review, removal)
+	if err != nil {
+		die(err.Error()) //nolint:govet
+	}
+
+	mm := make(map[string]string, len(userMeta)+len(backupMeta))
+	maps.Copy(mm, userMeta)
+	maps.Copy(mm, backupMeta)
+
+	return mm
+}
+
+// createBackupMetadata returns a map containing the backup metadata values if
+// the provided inputs are valid.
+func createBackupMetadata(reason, review, removal string) (map[string]string, error) {
+	mm := make(map[string]string, numOfBackupMeta)
+
+	reasonKey, err := put.ValidateAndCreateReasonMetadata(reason)
```

Comment:
Are the keys that these validate functions returns likely to change? You could just export it and reference it directly.